### ISSUE TITLE
[ FEATURE ] - Cache for Github API data 🎁

### ISSUE THREAD

**Issue by sachin-duhan26** [learn-opensource/learn-opensource.github.io#10]
To avoid the rate limit exceeding we are planning to implement a cache 💓  for storing data that is being retrieved previously from API.

## REQUIREMENT OVERVIEW 🖥️ 
```
js/main.js
```

1. [ ] Make a cache.js file manages & stores data in the browser storage [ local storage 🤫 ]
2. [ ] Update get call present in main.js and link it with newly developed cache mechanism.

### OTHER DETAILS
- Follow understandable & good naming convention 🗯️ 

This issue is **open for discussion** and we invite a better solution/way around to the same. 🎧

**Comment by leodeleone** [comment]
Hi! I'd like to work on this one. I guess we'll need an expiration time, right? How long should we consider it as valid?

**Comment by sachin-duhan26** [comment]
You are right @leodeleone , we can set an object with an expiration time in local storage.
- the period of expiration should be in a config file.
- value for the same may range from 2 days  - 1 week.

**Comment by sachin-duhan26** [comment]
hello @leodeleone , kindly check the code review.
we need to make a few changes for better implementation of code.
